### PR TITLE
[Merged by Bors] - Migrate to scale for transactions

### LINKED ISSUE #3271: codec: migrate to scale
Use https://github.com/spacemeshos/go-scale in the codec module.

before working on this issue implement https://github.com/spacemeshos/go-scale/issues/2

Using scale for majority of the objects should be trivial:
- add go generate with scalegen
- for every collection define sensible limit using tag name defined above
- add a fuzz tests and a golden tests (you should be able to find utilities for both in the lib above)

If some objects uses collections with pointers in them - they should be replaced with collections without pointers.

There is also some mess with types from poet/post, but i don't recall details.

### Modules to enable scale codec for:
- [x] transactions - https://github.com/spacemeshos/go-spacemesh/pull/3380
- [x] blocks - https://github.com/spacemeshos/go-spacemesh/pull/3377
- [x] ballots - https://github.com/spacemeshos/go-spacemesh/pull/3383
- [x] proposals - https://github.com/spacemeshos/go-spacemesh/pull/3383
- [x] fetch - https://github.com/spacemeshos/go-spacemesh/pull/3466
- [x] SQL proposal TX IDs - https://github.com/spacemeshos/go-spacemesh/pull/3448
- [x] poet proofs - https://github.com/spacemeshos/go-spacemesh/pull/3395
- [x] sync - https://github.com/spacemeshos/go-spacemesh/pull/3395
- [x] activation types - https://github.com/spacemeshos/go-spacemesh/pull/3413
- [x] hare - https://github.com/spacemeshos/go-spacemesh/pull/3478 
- [x] p2p server - https://github.com/spacemeshos/go-spacemesh/pull/3476
- [x] disable XDR codec + final fixes: https://github.com/spacemeshos/go-spacemesh/pull/3496, https://github.com/spacemeshos/go-spacemesh/pull/3482

### PR DESCRIPTION
**Pull request by evt**
## Motivation

Part of #3271 (for transactions)

## Changes
Applied scale codec to transactions

## Test Plan

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by evt** [comment]
bors try

**Review by countvonzero — COMMENTED** [review]
in the PR motivation, maybe don't use "Partly Closes". i suspect it will close the referenced issue when this PR is merged. 

maybe "part of #3380" or "partial implementation of #3380"?

UPDATE: yeah, your prev PR for block already closed #3271

File: common/types/transaction_header.go [hunk 1]
```diff
@@ -43,8 +45,6 @@ type LayerLimits struct {
 	Min, Max uint32
 }
 
-//go:generate scalegen -types Nonce
```
Comment (dshulyak) [inline common/types/transaction_header.go:46]: why it was removed?

**Review by dshulyak — APPROVED** [review]

**Comment by evt** [comment]
bors merge

**Comment by evt** [comment]
bors merge

**Pull request closed without merging**